File: t000414.py
def se(m, n):
    r = 0
    while m != 0:
        r += m % n
        m = int(m / n)
    return r
while 1:
    n = int(input())
    if n == 0: break
    if se(n, 10) == se(n, 16) == se(n, 12):
        print(str(n) + " is a Sky Number.")
    else:
        print(str(n) + " is not a Sky Number.")

File: t000414.cpp
#include <stdio.h>
 
int sum(int m ,int n)
{
	int result = 0;
	while (m != 0)
	{
		result += m % n;
		m /= n;
	}
	return result;
}
 
int main()
{
	int m;
	while (~scanf("%d",&m) && m)
	{
		if (sum(m,10) == sum(m,12) && sum(m,10) == sum(m,16))
		{
			printf("%d is a Sky Number.\n",m);
		}
		else
		{
			printf("%d is not a Sky Number.\n",m);
		}
 
	}
	return 0;
}
